packages/next/src/utils.ts

```diff
@@ -2567,10 +2512,13 @@ export function normalizeEdgeFunctionPath(
       appPathRoutesManifest[ogRoute] ||
       shortPath.replace(/(^|\/)(page|route)$/, '')
     ).replace(/^\//, '');
+  }
 
-    if (!shortPath || shortPath === '/') {
-      shortPath = 'index';
-    }
+  if (shortPath.startsWith('pages/')) {
+    shortPath = shortPath.replace(/^pages\//, '');
+  }
```

Comment:
```suggestion
  if (shortPath.startsWith('pages/')) {
    shortPath = shortPath.replace(/^pages\//, '');
  }
```

This can be removed now that its on line 2524